diff --git a/lib/commonjs/navigators/createNativeBottomTabNavigator.js b/lib/commonjs/navigators/createNativeBottomTabNavigator.js
index 347a7f64e232f3ab3f8048e45254b37542b9d6dd..5d7d839c9610ac71398fc5af7795b1fcff98a9cf 100644
--- a/lib/commonjs/navigators/createNativeBottomTabNavigator.js
+++ b/lib/commonjs/navigators/createNativeBottomTabNavigator.js
@@ -4,7 +4,7 @@ Object.defineProperty(exports, "__esModule", {
   value: true
 });
 exports.default = createNativeBottomTabNavigator;
-var _native = require("@react-navigation/native");
+var _native = require("expo-router/react-navigation");
 var _color = _interopRequireDefault(require("color"));
 var _NativeBottomTabView = _interopRequireDefault(require("../views/NativeBottomTabView.js"));
 var _jsxRuntime = require("react/jsx-runtime");
diff --git a/lib/commonjs/views/NativeBottomTabView.js b/lib/commonjs/views/NativeBottomTabView.js
index 9c646ab3d31489e1e5f0a432560fc1add86f6b5c..57a6726d798dbbd636b70e2a3d6f525c6c54622f 100644
--- a/lib/commonjs/views/NativeBottomTabView.js
+++ b/lib/commonjs/views/NativeBottomTabView.js
@@ -4,7 +4,7 @@ Object.defineProperty(exports, "__esModule", {
   value: true
 });
 exports.default = NativeBottomTabView;
-var _native = require("@react-navigation/native");
+var _native = require("expo-router/react-navigation");
 var _reactNativeBottomTabs = _interopRequireDefault(require("react-native-bottom-tabs"));
 var _jsxRuntime = require("react/jsx-runtime");
 function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
diff --git a/lib/module/navigators/createNativeBottomTabNavigator.js b/lib/module/navigators/createNativeBottomTabNavigator.js
index 6e6c05312ccb0697474678f01e5f832d1f66f778..f9b49cb2806715b82f5814dca65cfaa77a2d169b 100644
--- a/lib/module/navigators/createNativeBottomTabNavigator.js
+++ b/lib/module/navigators/createNativeBottomTabNavigator.js
@@ -1,6 +1,6 @@
 "use strict";
 
-import { createNavigatorFactory, TabRouter, useNavigationBuilder, useTheme } from '@react-navigation/native';
+import { createNavigatorFactory, TabRouter, useNavigationBuilder, useTheme } from 'expo-router/react-navigation';
 import Color from 'color';
 import NativeBottomTabView from "../views/NativeBottomTabView.js";
 import { jsx as _jsx } from "react/jsx-runtime";
diff --git a/lib/module/views/NativeBottomTabView.js b/lib/module/views/NativeBottomTabView.js
index c824cd1b65331a027ed7042a6939bf9d9a694e01..88a9e3b5350a9b0ed9bce28c4d4f581d620dbc5a 100644
--- a/lib/module/views/NativeBottomTabView.js
+++ b/lib/module/views/NativeBottomTabView.js
@@ -1,6 +1,6 @@
 "use strict";
 
-import { CommonActions } from '@react-navigation/native';
+import { CommonActions } from 'expo-router/react-navigation';
 import TabView from 'react-native-bottom-tabs';
 import { jsx as _jsx } from "react/jsx-runtime";
 export default function NativeBottomTabView({
diff --git a/lib/typescript/commonjs/src/navigators/createNativeBottomTabNavigator.d.ts b/lib/typescript/commonjs/src/navigators/createNativeBottomTabNavigator.d.ts
index 5ef6162dd7db44aad647cc6ae60ec28177f24a46..8ff8e9fe3a92629e49ddb54bfc9c7976ac21fdee 100644
--- a/lib/typescript/commonjs/src/navigators/createNativeBottomTabNavigator.d.ts
+++ b/lib/typescript/commonjs/src/navigators/createNativeBottomTabNavigator.d.ts
@@ -1,4 +1,4 @@
-import { type DefaultNavigatorOptions, type NavigatorTypeBagBase, type ParamListBase, type StaticConfig, type TabNavigationState, type TabRouterOptions, type TypedNavigator } from '@react-navigation/native';
+import { type DefaultNavigatorOptions, type NavigatorTypeBagBase, type ParamListBase, type StaticConfig, type TabNavigationState, type TabRouterOptions, type TypedNavigator } from 'expo-router/react-navigation';
 import type { NativeBottomTabNavigationConfig, NativeBottomTabNavigationEventMap, NativeBottomTabNavigationOptions, NativeBottomTabNavigationProp } from '../types';
 export type NativeBottomTabNavigatorProps = DefaultNavigatorOptions<ParamListBase, string | undefined, TabNavigationState<ParamListBase>, NativeBottomTabNavigationOptions, NativeBottomTabNavigationEventMap, NativeBottomTabNavigationProp<ParamListBase>> & TabRouterOptions & NativeBottomTabNavigationConfig;
 declare function NativeBottomTabNavigator({ id, initialRouteName, backBehavior, UNSTABLE_routeNamesChangeBehavior, children, layout, screenListeners, screenOptions, screenLayout, UNSTABLE_router, tabBarActiveTintColor: customActiveTintColor, tabBarInactiveTintColor: customInactiveTintColor, layoutDirection, ...rest }: NativeBottomTabNavigatorProps): import("react/jsx-runtime").JSX.Element;
diff --git a/lib/typescript/commonjs/src/types.d.ts b/lib/typescript/commonjs/src/types.d.ts
index 3cd0887352379be69de67d348be2686baad80c87..df5ae5fe1366b40707e41a86b2a34bdca16857d7 100644
--- a/lib/typescript/commonjs/src/types.d.ts
+++ b/lib/typescript/commonjs/src/types.d.ts
@@ -1,4 +1,4 @@
-import type { Descriptor, NavigationHelpers, NavigationProp, ParamListBase, RouteProp, TabActionHelpers, TabNavigationState } from '@react-navigation/native';
+import type { Descriptor, NavigationHelpers, NavigationProp, ParamListBase, RouteProp, TabActionHelpers, TabNavigationState } from 'expo-router/react-navigation';
 import type { ImageSourcePropType, StyleProp, ViewStyle } from 'react-native';
 import type TabView from 'react-native-bottom-tabs';
 import type { AppleIcon, TabRole } from 'react-native-bottom-tabs';
diff --git a/lib/typescript/commonjs/src/views/NativeBottomTabView.d.ts b/lib/typescript/commonjs/src/views/NativeBottomTabView.d.ts
index df1cb50317b95d97fcc047bb403bed7a4c194e48..dceab7bfc2811182ffb00bf8631709c0100eb95d 100644
--- a/lib/typescript/commonjs/src/views/NativeBottomTabView.d.ts
+++ b/lib/typescript/commonjs/src/views/NativeBottomTabView.d.ts
@@ -1,4 +1,4 @@
-import { type ParamListBase, type TabNavigationState } from '@react-navigation/native';
+import { type ParamListBase, type TabNavigationState } from 'expo-router/react-navigation';
 import type { NativeBottomTabDescriptorMap, NativeBottomTabNavigationConfig, NativeBottomTabNavigationHelpers } from '../types';
 type Props = NativeBottomTabNavigationConfig & {
     state: TabNavigationState<ParamListBase>;
diff --git a/lib/typescript/module/src/navigators/createNativeBottomTabNavigator.d.ts b/lib/typescript/module/src/navigators/createNativeBottomTabNavigator.d.ts
index 5ef6162dd7db44aad647cc6ae60ec28177f24a46..8ff8e9fe3a92629e49ddb54bfc9c7976ac21fdee 100644
--- a/lib/typescript/module/src/navigators/createNativeBottomTabNavigator.d.ts
+++ b/lib/typescript/module/src/navigators/createNativeBottomTabNavigator.d.ts
@@ -1,4 +1,4 @@
-import { type DefaultNavigatorOptions, type NavigatorTypeBagBase, type ParamListBase, type StaticConfig, type TabNavigationState, type TabRouterOptions, type TypedNavigator } from '@react-navigation/native';
+import { type DefaultNavigatorOptions, type NavigatorTypeBagBase, type ParamListBase, type StaticConfig, type TabNavigationState, type TabRouterOptions, type TypedNavigator } from 'expo-router/react-navigation';
 import type { NativeBottomTabNavigationConfig, NativeBottomTabNavigationEventMap, NativeBottomTabNavigationOptions, NativeBottomTabNavigationProp } from '../types';
 export type NativeBottomTabNavigatorProps = DefaultNavigatorOptions<ParamListBase, string | undefined, TabNavigationState<ParamListBase>, NativeBottomTabNavigationOptions, NativeBottomTabNavigationEventMap, NativeBottomTabNavigationProp<ParamListBase>> & TabRouterOptions & NativeBottomTabNavigationConfig;
 declare function NativeBottomTabNavigator({ id, initialRouteName, backBehavior, UNSTABLE_routeNamesChangeBehavior, children, layout, screenListeners, screenOptions, screenLayout, UNSTABLE_router, tabBarActiveTintColor: customActiveTintColor, tabBarInactiveTintColor: customInactiveTintColor, layoutDirection, ...rest }: NativeBottomTabNavigatorProps): import("react/jsx-runtime").JSX.Element;
diff --git a/lib/typescript/module/src/types.d.ts b/lib/typescript/module/src/types.d.ts
index 3cd0887352379be69de67d348be2686baad80c87..df5ae5fe1366b40707e41a86b2a34bdca16857d7 100644
--- a/lib/typescript/module/src/types.d.ts
+++ b/lib/typescript/module/src/types.d.ts
@@ -1,4 +1,4 @@
-import type { Descriptor, NavigationHelpers, NavigationProp, ParamListBase, RouteProp, TabActionHelpers, TabNavigationState } from '@react-navigation/native';
+import type { Descriptor, NavigationHelpers, NavigationProp, ParamListBase, RouteProp, TabActionHelpers, TabNavigationState } from 'expo-router/react-navigation';
 import type { ImageSourcePropType, StyleProp, ViewStyle } from 'react-native';
 import type TabView from 'react-native-bottom-tabs';
 import type { AppleIcon, TabRole } from 'react-native-bottom-tabs';
diff --git a/lib/typescript/module/src/views/NativeBottomTabView.d.ts b/lib/typescript/module/src/views/NativeBottomTabView.d.ts
index df1cb50317b95d97fcc047bb403bed7a4c194e48..dceab7bfc2811182ffb00bf8631709c0100eb95d 100644
--- a/lib/typescript/module/src/views/NativeBottomTabView.d.ts
+++ b/lib/typescript/module/src/views/NativeBottomTabView.d.ts
@@ -1,4 +1,4 @@
-import { type ParamListBase, type TabNavigationState } from '@react-navigation/native';
+import { type ParamListBase, type TabNavigationState } from 'expo-router/react-navigation';
 import type { NativeBottomTabDescriptorMap, NativeBottomTabNavigationConfig, NativeBottomTabNavigationHelpers } from '../types';
 type Props = NativeBottomTabNavigationConfig & {
     state: TabNavigationState<ParamListBase>;
diff --git a/lib/typescript/src/navigators/createNativeBottomTabNavigator.d.ts b/lib/typescript/src/navigators/createNativeBottomTabNavigator.d.ts
index 141f7894c467d322cdcc61dae81d3167f190eecb..b96ac42118f7db37cf461d78811c709b26b30979 100644
--- a/lib/typescript/src/navigators/createNativeBottomTabNavigator.d.ts
+++ b/lib/typescript/src/navigators/createNativeBottomTabNavigator.d.ts
@@ -1,4 +1,4 @@
-import { type DefaultNavigatorOptions, type NavigatorTypeBagBase, type ParamListBase, type StaticConfig, type TabNavigationState, type TabRouterOptions, type TypedNavigator } from '@react-navigation/native';
+import { type DefaultNavigatorOptions, type NavigatorTypeBagBase, type ParamListBase, type StaticConfig, type TabNavigationState, type TabRouterOptions, type TypedNavigator } from 'expo-router/react-navigation';
 import type { NativeBottomTabNavigationConfig, NativeBottomTabNavigationEventMap, NativeBottomTabNavigationOptions, NativeBottomTabNavigationProp } from '../types';
 export type NativeBottomTabNavigatorProps = DefaultNavigatorOptions<ParamListBase, string | undefined, TabNavigationState<ParamListBase>, NativeBottomTabNavigationOptions, NativeBottomTabNavigationEventMap, NativeBottomTabNavigationProp<ParamListBase>> & TabRouterOptions & NativeBottomTabNavigationConfig;
 declare function NativeBottomTabNavigator({ id, initialRouteName, backBehavior, UNSTABLE_routeNamesChangeBehavior, children, layout, screenListeners, screenOptions, screenLayout, UNSTABLE_router, tabBarActiveTintColor: customActiveTintColor, tabBarInactiveTintColor: customInactiveTintColor, layoutDirection, ...rest }: NativeBottomTabNavigatorProps): import("react/jsx-runtime").JSX.Element;
diff --git a/lib/typescript/src/navigators/createNativeBottomTabNavigator.js b/lib/typescript/src/navigators/createNativeBottomTabNavigator.js
index 3a32a5fb837687abf1b6b67e987de2827e410fe5..d78d58d968b825bf06ebcce614d7fb58553c57e6 100644
--- a/lib/typescript/src/navigators/createNativeBottomTabNavigator.js
+++ b/lib/typescript/src/navigators/createNativeBottomTabNavigator.js
@@ -1,5 +1,5 @@
 import { jsx as _jsx } from "react/jsx-runtime";
-import { createNavigatorFactory, TabRouter, useNavigationBuilder, useTheme, } from '@react-navigation/native';
+import { createNavigatorFactory, TabRouter, useNavigationBuilder, useTheme, } from 'expo-router/react-navigation';
 import Color from 'color';
 import NativeBottomTabView from '../views/NativeBottomTabView';
 function NativeBottomTabNavigator({ id, initialRouteName, backBehavior, UNSTABLE_routeNamesChangeBehavior, children, layout, screenListeners, screenOptions, screenLayout, UNSTABLE_router, tabBarActiveTintColor: customActiveTintColor, tabBarInactiveTintColor: customInactiveTintColor, layoutDirection = 'locale', ...rest }) {
diff --git a/lib/typescript/src/types.d.ts b/lib/typescript/src/types.d.ts
index 1034d88d35f7ee56032972f18a9e255f8be441ac..31c3bb80bd18d7e7a2f19c49be448accfb64ba0d 100644
--- a/lib/typescript/src/types.d.ts
+++ b/lib/typescript/src/types.d.ts
@@ -1,4 +1,4 @@
-import type { Descriptor, NavigationHelpers, NavigationProp, ParamListBase, RouteProp, TabActionHelpers, TabNavigationState } from '@react-navigation/native';
+import type { Descriptor, NavigationHelpers, NavigationProp, ParamListBase, RouteProp, TabActionHelpers, TabNavigationState } from 'expo-router/react-navigation';
 import type { ImageSourcePropType, StyleProp, ViewStyle } from 'react-native';
 import type TabView from 'react-native-bottom-tabs';
 import type { AppleIcon, TabRole } from 'react-native-bottom-tabs';
diff --git a/lib/typescript/src/views/NativeBottomTabView.d.ts b/lib/typescript/src/views/NativeBottomTabView.d.ts
index 3e1a4fe8627e4f964d9e50811865481fc9a15621..9ef8f38e285f9e0e1c01289f4538e735b38e41ff 100644
--- a/lib/typescript/src/views/NativeBottomTabView.d.ts
+++ b/lib/typescript/src/views/NativeBottomTabView.d.ts
@@ -1,4 +1,4 @@
-import { type ParamListBase, type TabNavigationState } from '@react-navigation/native';
+import { type ParamListBase, type TabNavigationState } from 'expo-router/react-navigation';
 import type { NativeBottomTabDescriptorMap, NativeBottomTabNavigationConfig, NativeBottomTabNavigationHelpers } from '../types';
 type Props = NativeBottomTabNavigationConfig & {
     state: TabNavigationState<ParamListBase>;
diff --git a/lib/typescript/src/views/NativeBottomTabView.js b/lib/typescript/src/views/NativeBottomTabView.js
index cd71e918e3f62b12f07a319d1aace80f63c5173c..ed2cafea11bfaec73d9396447b34f0b80f8d0005 100644
--- a/lib/typescript/src/views/NativeBottomTabView.js
+++ b/lib/typescript/src/views/NativeBottomTabView.js
@@ -1,5 +1,5 @@
 import { jsx as _jsx } from "react/jsx-runtime";
-import { CommonActions, } from '@react-navigation/native';
+import { CommonActions, } from 'expo-router/react-navigation';
 import TabView from 'react-native-bottom-tabs';
 export default function NativeBottomTabView({ state, layoutDirection, navigation, descriptors, tabBar, ...rest }) {
     return (_jsx(TabView, { ...rest, navigationState: state, renderScene: ({ route }) => descriptors[route.key]?.render(), getActiveTintColor: ({ route }) => {
diff --git a/src/navigators/createNativeBottomTabNavigator.tsx b/src/navigators/createNativeBottomTabNavigator.tsx
index 58f1647828094ca749ed3de9dae6bc924683ebf5..f2ba1e5780f8407622c766a8c27a8e09ee46a483 100644
--- a/src/navigators/createNativeBottomTabNavigator.tsx
+++ b/src/navigators/createNativeBottomTabNavigator.tsx
@@ -11,7 +11,7 @@ import {
   type TypedNavigator,
   useNavigationBuilder,
   useTheme,
-} from '@react-navigation/native';
+} from 'expo-router/react-navigation';
 import Color from 'color';
 
 import type {
diff --git a/src/types.ts b/src/types.ts
index b55afb0563b488e24adc652213448c28e3c3aea8..5d94db4fb387bc701ac8740db6e252695093a64b 100644
--- a/src/types.ts
+++ b/src/types.ts
@@ -6,7 +6,7 @@ import type {
   RouteProp,
   TabActionHelpers,
   TabNavigationState,
-} from '@react-navigation/native';
+} from 'expo-router/react-navigation';
 import type { ImageSourcePropType, StyleProp, ViewStyle } from 'react-native';
 import type TabView from 'react-native-bottom-tabs';
 import type { AppleIcon, TabRole } from 'react-native-bottom-tabs';
diff --git a/src/views/NativeBottomTabView.tsx b/src/views/NativeBottomTabView.tsx
index 787cc5b5549ccc539f83135022c91a721c7166b1..77cda1bb6fe2d75df096c38996ea731730a63a61 100644
--- a/src/views/NativeBottomTabView.tsx
+++ b/src/views/NativeBottomTabView.tsx
@@ -3,7 +3,7 @@ import {
   type TabNavigationState,
   type Route,
   CommonActions,
-} from '@react-navigation/native';
+} from 'expo-router/react-navigation';
 import type {
   NativeBottomTabDescriptorMap,
   NativeBottomTabNavigationConfig,
