diff --git a/android/build.gradle.kts b/android/build.gradle.kts
index e326d97ca156b074b1f5a41a310edef191695e56..4cefb40e8225fa7e8f8e0ec60c8fadf4deb0661d 100644
--- a/android/build.gradle.kts
+++ b/android/build.gradle.kts
@@ -322,7 +322,8 @@ tasks.named("preBuild") { dependsOn("assertMinimalReactNativeVersionTask") }
 
 // Workaround for AGP 9 + Kotlin 2.x lint K2 UAST crash on .gradle.kts build scripts.
 // See: https://issuetracker.google.com/issues/432144179
-tasks.configureEach { if (name.startsWith("lintVital")) enabled = false }
+// Backport https://github.com/software-mansion/react-native-reanimated/pull/10448.
+tasks.configureEach { if (name.startsWith("lint")) enabled = false }
 
 tasks.register<Copy>("prepareWorkletsHeadersForPrefabs") {
     from("$projectDir/src/main/cpp")
